glue_astronomy/translators/regions.py

```diff
@@ -46,6 +48,36 @@ def range_to_rect(data, ori, low, high):
     return RectanglePixelRegion(PixCoord(xcen, ycen), width, height)
 
 
+def _is_annulus(subset_state):
+    # subset_state.state1 = outer circle
+    # subset_state.state2 = inner circle
+    # subset_state.state2 is inverted, so we need its state1
+    return ((not isinstance(subset_state.state1, InvertState)) and
+            isinstance(subset_state.state1.roi, CircularROI) and
+            isinstance(subset_state.state2, InvertState) and
+            isinstance(subset_state.state2.state1.roi, CircularROI) and
+            (subset_state.state1.roi.xc == subset_state.state2.state1.roi.xc) and
+            (subset_state.state1.roi.yc == subset_state.state2.state1.roi.yc) and
+            (subset_state.state1.roi.radius > subset_state.state2.state1.roi.radius))
+
+
+# Put this here because there is nowhere else to put it.
+# https://github.com/glue-viz/glue/issues/2390
+def _annulus_to_subset_state(reg, data):
+    """CircleAnnulusPixelRegion to glue subset state."""
+    # TODO: Add ellipse and rectangle support.
+    if not isinstance(reg, CircleAnnulusPixelRegion):  # pragma: no cover
+        raise NotImplementedError(f"{reg} not supported")
```

Comment:
```suggestion
def _region_to_subset_state(reg, data):
    """CircleAnnulusPixelRegion to glue subset state."""
    if not isinstance(reg, PixelRegion):
        raise ValueError("`reg` should be a `PixelRegion` instance")
    # TODO: Add ellipse and rectangle support.
    elif not isinstance(reg, CircleAnnulusPixelRegion):  # pragma: no cover
        raise NotImplementedError(f"{reg} not supported")
```
If this should/will be extended to other [Pixel?]Region classes, maybe better to chose a more general name right away?

Reply:
I don't think there is any need. Other simpler regions are doing the ROI thing. Annulus is complicated because it is a composite subset with 2 ROIs and very rigid rules.

Reply:
Then remove the TODO?

Reply:
Oh, sorry, I mean elliptical annulus and rectangular annulus. We don't have a need for that right now, so I don't want to waste time implementing but they would also go into the same function.

Reply:
Ah, I did not even know those existed! Then the name makes sense; perhaps just add the modified suggestion (because the first checks what will never be supported, the second only what's not implemented yet)?

Reply:
I accepted your suggestion. Thanks!